### PR TITLE
[14.0][IMP] purchase_stock_picking_invoice_link: Use the same technique as in stock_picking_invoice_link

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT38622

FW-PORT of:

- https://github.com/OCA/stock-logistics-workflow/commit/3d1490ef091c291e5edf41a9037e0189cede570e
- https://github.com/OCA/stock-logistics-workflow/commit/a6b1bc4631e597687664dcfbc1a6a03ba55a48a7

please @cesar-tecnativa @pedrobaeza review this

### CODE REVIEW

File: purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py [hunk 1]
```diff
@@ -66,3 +66,63 @@ def test_link_with_on_receive_quantities_policy(self):
...
+        # Refund invoice
+        wiz_invoice_refund = (
+            self.env["account.move.reversal"]
+            .with_context(active_model="account.move", active_ids=invoice.ids)
+            .create({"refund_method": "modify", "reason": "test"})
+        )
+        invoice_id = wiz_invoice_refund.reverse_moves()["res_id"]
+        new_inv = self.env["account.move"].browse(invoice_id)
+        new_inv.invoice_date = self.po.create_date
+        new_inv.action_post()
```
Comment (pedrobaeza) [inline purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py]: No need of this post for saving a bit of time

File: purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py [hunk 2]
```diff
@@ -66,3 +66,63 @@ def test_link_with_on_receive_quantities_policy(self):
...
+        invoice.action_post()
+        # Refund invoice
+        wiz_invoice_refund = (
+            self.env["account.move.reversal"]
+            .with_context(active_model="account.move", active_ids=invoice.ids)
+            .create({"refund_method": "modify", "reason": "test"})
+        )
+        invoice_id = wiz_invoice_refund.reverse_moves()["res_id"]
+        new_inv = self.env["account.move"].browse(invoice_id)
+        new_inv.invoice_date = self.po.create_date
```
Comment (pedrobaeza) [inline purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py]: Nor this

File: purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py [hunk 3]
```diff
@@ -66,3 +66,63 @@ def test_link_with_on_receive_quantities_policy(self):
...
+        new_inv.action_post()
+        self.assertTrue(len(picking.invoice_ids) == 3)
```
Comment (pedrobaeza) [inline purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py]: Why not?

```suggestion
        self.assertEqual(len(picking.invoice_ids), 3)
```

**Comment by CarlosRoca13** [comment]
@pedrobaeza changes done

File: purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py [hunk 4]
```diff
@@ -66,3 +66,64 @@ def test_link_with_on_receive_quantities_policy(self):
...
+            self.env["account.move.reversal"]
+            .with_context(active_model="account.move", active_ids=invoice.ids)
+            .create({"refund_method": "cancel", "reason": "test"})
+        )
+        wiz_invoice_refund.reverse_moves()
+        # Create invoice again
+        inv_action = self.po.action_create_invoice()
+        new_inv = self.env["account.move"].browse([(inv_action["res_id"])])
+        new_inv.invoice_date = self.po.create_date
+        new_inv.action_post()
```
Comment (pedrobaeza) [inline purchase_stock_picking_invoice_link/tests/test_purchase_stock_picking_invoice_link.py]: Don't assign date nor post here as well

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@rvalyi this one is the twin PR.

File: purchase_stock_picking_invoice_link/models/purchase_order.py [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# Copyright 2022 Tecnativa - Carlos Roca
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models
+from odoo.tools import float_compare, float_is_zero
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "purchase.order.line"
```
Comment (rvalyi) [inline purchase_stock_picking_invoice_link/models/purchase_order.py:8]: @pedrobaeza too much twin: should be class PurchaseOrderLine(models.Model):

**Review by rvalyi — CHANGES_REQUESTED** [review]

File: purchase_stock_picking_invoice_link/models/purchase_order.py (hunk 5 above)
Comment (pedrobaeza) [reply purchase_stock_picking_invoice_link/models/purchase_order.py:8]: Hehe, indeed. Although it doesn't affect on Odoo itself, let's name it correctly.

Comment (pedrobaeza) [reply purchase_stock_picking_invoice_link/models/purchase_order.py:8]: @rvalyi changed

**Review by rvalyi — APPROVED** [review]
great!

**Pull request merged**